src/VirtoCommerce.Platform.Assets.AzureBlobStorage/AzureBlobProvider.cs

```diff
@@ -153,114 +156,114 @@ public virtual async Task<BlobEntrySearchResult> SearchAsync(string folderUrl, s
 
             if (!string.IsNullOrEmpty(folderUrl))
             {
-                var blobContainer = GetBlobContainer(GetContainerNameFromUrl(folderUrl));
+                var container = GetBlobContainer(GetContainerNameFromUrl(folderUrl));
 
-                if (blobContainer != null)
+                if (container != null)
                 {
-                    BlobContinuationToken blobContinuationToken = null;
-
-                    var directoryPath = GetDirectoryPathFromUrl(folderUrl);
-                    var blobDirectory = !string.IsNullOrEmpty(directoryPath)
-                        ? blobContainer.GetDirectoryReference(directoryPath)
-                        : null;
-                    var listBlobs = blobDirectory != null
-                        ? await blobDirectory.ListBlobsSegmentedAsync(blobContinuationToken)
-                        : await blobContainer.ListBlobsSegmentedAsync(null, blobContinuationToken);
-                    if (!string.IsNullOrEmpty(keyword))
+                    var baseUri = container.Uri.AbsoluteUri;
+                    var prefix = GetDirectoryPathFromUrl(folderUrl);
+                    if (prefix != null && !string.IsNullOrEmpty(keyword))
                     {
-                        if (blobDirectory != null)
-                        {
-                            keyword = blobDirectory.Prefix + keyword;
-                        }
-
-                        BlobContinuationToken prefixBlobContinuationToken = null;
                         //Only whole container list allow search by prefix
-                        listBlobs = await blobContainer.ListBlobsSegmentedAsync(keyword, prefixBlobContinuationToken);
+                        prefix += keyword;
                     }
 
-                    // Loop over items within the container and output the length and URI.
-                    foreach (IListBlobItem item in listBlobs.Results)
+                    // Call the listing operation and return pages of the specified size.
+                    var resultSegment = container.GetBlobsByHierarchyAsync(prefix: prefix, delimiter: Delimiter)
+                        .AsPages();
+
+                    // Enumerate the blobs returned for each page.
+                    await foreach (var blobPage in resultSegment)
                     {
-                        if (item is CloudBlockBlob block)
+                        // A hierarchical listing may return both virtual directories and blobs.
+                        foreach (var blobhierarchyItem in blobPage.Values)
                         {
-                            var blobInfo = ConvertBlobToBlobInfo(block);
-                            //Do not return empty blob (created with directory because azure blob not support direct directory creation)
-                            if (!string.IsNullOrEmpty(blobInfo.Name))
+                            if (blobhierarchyItem.IsPrefix)
```

Comment:
Missed Uri.UnescapeDataString() for folder Names and Url, also missed folder.ParentUrl setting

Reply:
Uri.EscapeUriString added for urls. 
Uri.UnescapeDataString is needed only for Urls, but not for 
`folder.Name = blobhierarchyItem.Prefix`